src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -186,22 +190,14 @@ public void insert(Collection<SinkRecord> records) {
       if (recordService.shouldSkipNullValue(record, behaviorOnNullValues)) {
         continue;
       }
-      // Might happen a count of record based flushing
+      // While inserting into buffer, we will check for count threshold and buffered bytes
+      // threshold.
       insert(record);
     }
-    // check all sink context to see if they need to be flushed
+    // check all partitions to see if they need to be flushed based on time
     for (TopicPartitionChannel partitionChannel : partitionsToChannel.values()) {
       // Time based flushing
-      if ((System.currentTimeMillis() - partitionChannel.getPreviousFlushTimeStampMs())
-          >= (getFlushTime() * 1000)) {
-        LOGGER.info("Time based flush for channel:{}", partitionChannel.getChannelName());
-        LOGGER.info(
-            "Current:{}, previousFlushTime:{}, threshold:{}",
-            System.currentTimeMillis(),
-            partitionChannel.getPreviousFlushTimeStampMs(),
-            (getFlushTime() * 1000));
-        partitionChannel.insertBufferedRows();
-      }
+      partitionChannel.insertBufferedRowsIfFlushTimeThresholdReached();
```

Comment:
I believe we have a TODO somewhere to make this check happens in a separate thread?